code/framework/extension/src/main/java/io/cattle/platform/extension/impl/ExtensionManagerImpl.java

```diff
@@ -61,6 +67,21 @@
     }
 
     @Override
+    public Map<String, Object> map(String key) {
+        return getExtensionMapInternal(key);
+    }
+
+    protected synchronized ExtensionMap<String, Object> getExtensionMapInternal(String key) {
+        ExtensionMap<String, Object> map = extensionMaps.get(key);
+        if (map == null) {
+            Map<String, Object> inner = started ? getMap(key) : Collections.<String, Object>emptyMap();
```

Comment:
The inner map does not need to be a synchronized collection.  In fact it shouldn't be.